cibyl/models/ci/system.py

```diff
@@ -78,7 +90,7 @@ def __str__(self, indent=0, verbosity=0):
 
     def populate(self, instances_dict):
         """Populate instances from a given dictionary of instances."""
-        if instances_dict.attr_type == Job:
+        if instances_dict.attr_type == self.top_level_model:
             for job in instances_dict.values():
```

Comment:
So if I understand correctly this is no long reference Job necessarily, so probably for model_instance in instances_dict...?

Reply:
I don't think this has changed, the point of this check for me was to ensure that we only introduced Job objects in the populate step, and nothing has changed in this regard.

Reply:
but self.top_level_model can be a pipeline, no?  so we we use specifically the word `job` in ` for job in instances_dict.values():`?

Reply:
Yes, you're I right, I misread you original comment, that is changed in the latest commit